newIDE/app/src/ProjectManager/ProjectPropertiesDialog.js

```diff
@@ -46,417 +47,400 @@ type State = {|
   isFolderProject: boolean,
 |};
 
-class ProjectPropertiesDialog extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = this._loadFrom(props.project);
-  }
+function loadPropertiesFromProject(project: gdProject): ProjectProperties {
+  return {
+    gameResolutionWidth: project.getGameResolutionWidth(),
+    gameResolutionHeight: project.getGameResolutionHeight(),
+    adaptGameResolutionAtRuntime: project.getAdaptGameResolutionAtRuntime(),
+    name: project.getName(),
+    author: project.getAuthor(),
+    version: project.getVersion(),
+    packageName: project.getPackageName(),
+    orientation: project.getOrientation(),
+    scaleMode: project.getScaleMode(),
+    sizeOnStartupMode: project.getSizeOnStartupMode(),
+    showGDevelopSplash: project.getLoadingScreen().isGDevelopSplashShown(),
+    minFPS: project.getMinimumFPS(),
+    maxFPS: project.getMaximumFPS(),
+    isFolderProject: project.isFolderProject(),
+  };
+}
 
-  _subscriptionChecker: ?SubscriptionChecker = null;
+function applyPropertiesToProject(
+  project: gdProject,
+  newProperties: ProjectProperties
+) {
+  const t = str => str; //TODO
+  const {
+    gameResolutionWidth,
+    gameResolutionHeight,
+    adaptGameResolutionAtRuntime,
+    name,
+    author,
+    version,
+    packageName,
+    orientation,
+    scaleMode,
+    sizeOnStartupMode,
+    showGDevelopSplash,
+    minFPS,
+    maxFPS,
+    isFolderProject,
+  } = newProperties;
+  project.setGameResolutionSize(gameResolutionWidth, gameResolutionHeight);
+  project.setAdaptGameResolutionAtRuntime(adaptGameResolutionAtRuntime);
+  project.setName(name);
+  project.setAuthor(author);
+  project.setVersion(version);
+  project.setPackageName(packageName);
+  project.setOrientation(orientation);
+  project.setScaleMode(scaleMode);
+  project.setSizeOnStartupMode(sizeOnStartupMode);
+  project.setMinimumFPS(minFPS);
+  project.setMaximumFPS(maxFPS);
+  project.getLoadingScreen().showGDevelopSplash(showGDevelopSplash);
+  project.setFolderProject(isFolderProject);
 
-  _loadFrom(project: gdProject): State {
-    return {
-      gameResolutionWidth: project.getGameResolutionWidth(),
-      gameResolutionHeight: project.getGameResolutionHeight(),
-      adaptGameResolutionAtRuntime: project.getAdaptGameResolutionAtRuntime(),
-      name: project.getName(),
-      author: project.getAuthor(),
-      version: project.getVersion(),
-      packageName: project.getPackageName(),
-      orientation: project.getOrientation(),
-      adMobAppId: project.getAdMobAppId(),
-      scaleMode: project.getScaleMode(),
-      sizeOnStartupMode: project.getSizeOnStartupMode(),
-      showGDevelopSplash: project.getLoadingScreen().isGDevelopSplashShown(),
-      minFPS: project.getMinimumFPS(),
-      maxFPS: project.getMaximumFPS(),
-      isFolderProject: project.isFolderProject(),
-    };
-  }
+  if (!displayProjectErrorsBox(t, getErrors(t, project))) return;
```

Comment:
This should be preventing "Apply" to finish if there are errors.
For this: `return displayProjectErrorsBox(...)` and when calling `applyPropertiesToProject`, don't call `props.onApply();` if false is returned.